### PR TITLE
Bump solana-rbpf crate to 0.5.0

### PR DESCRIPTION
**Pull request by seanyoung**
This makes it possible to use rust 1.70.

### CODE REVIEW

File: tests/solana.rs [hunk 1]
```diff
@@ -1017,40 +1017,37 @@ fn translate(
...
-    unsafe {
-        match translate(memory_mapping, access_type, vm_addr, size_of::<T>() as u64) {
-            ProgramResult::Ok(value) => Ok(&mut *(value as *mut T)),
-            ProgramResult::Err(e) => Err(e),
-        }
-    }
+) -> Result<&'a mut T, Error> {
+    let host_addr = translate(memory_mapping, access_type, vm_addr, size_of::<T>() as u64)?;
+
+    Ok(unsafe { &mut *(host_addr as *mut T) })
```
Comment (xermicus) [inline tests/solana.rs:1031]: I guess in the test it's not terribly important. But usually `unsafe {}` block come with a comment explaining their safety assumptions.

File: tests/solana.rs [hunk 2]
```diff
@@ -1059,28 +1056,19 @@ fn translate_slice_inner<'a, T>(
...
-            ProgramResult::Err(e) => Err(e),
-        }
+        return Ok(&mut []);
     }
+
+    let total_size = len.saturating_mul(size_of::<T>() as u64);
+
+    let host_addr = translate(memory_mapping, access_type, vm_addr, total_size)?;
+
+    Ok(unsafe { std::slice::from_raw_parts_mut(host_addr as *mut T, len as usize) })
```
Comment (xermicus) [inline tests/solana.rs:1068]: Same

**Review by xermicus — APPROVED** [review]

**Review by LucasSte — APPROVED** [review]

File: tests/solana.rs (hunk 1 above)
Comment (seanyoung) [reply tests/solana.rs:1033]: I've added a comment

File: tests/solana.rs (hunk 2 above)
Comment (seanyoung) [reply tests/solana.rs:1071]: I've added a comment

**Pull request merged**